Remove build warning

The current code base does not build cleanly any more:

```
...
[ 62%] Building C object test/CMakeFiles/oqs_test_groups.dir/tlstest_helpers.c.o
~/oqs-provider/test/tlstest_helpers.c: In function ‘create_cert_key’:
~/oqs-provider/test/tlstest_helpers.c:25:48: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   25 |         !X509_set_pubkey(x509, pkey) || !(name = X509_get_subject_name(x509)) ||
      |          
...                                      ^
```

--> `openssl` API for "X509_get_subject_name" was const-ified, so the `oqsprovider` code should adapt to that API change (without breaking in the 3.x APIs, of course).

--- a/test/tlstest_helpers.c
+++ b/test/tlstest_helpers.c
@@ -22,7 +22,8 @@ int create_cert_key(OSSL_LIB_CTX *libctx, char *algname, char *certfilename,
         !ASN1_INTEGER_set(X509_get_serialNumber(x509), 1) ||
         !X509_gmtime_adj(X509_getm_notBefore(x509), 0) ||
         !X509_gmtime_adj(X509_getm_notAfter(x509), 31536000L) ||
-        !X509_set_pubkey(x509, pkey) || !(name = X509_get_subject_name(x509)) ||
+        !X509_set_pubkey(x509, pkey) ||
+        !(name = X509_NAME_dup(X509_get_subject_name(x509))) ||
         !X509_NAME_add_entry_by_txt(name, "C", MBSTRING_ASC,
                                     (unsigned char *)"CH", -1, -1, 0) ||
         !X509_NAME_add_entry_by_txt(name, "O", MBSTRING_ASC,
@@ -39,6 +40,7 @@ int create_cert_key(OSSL_LIB_CTX *libctx, char *algname, char *certfilename,
 
     EVP_PKEY_free(pkey);
     X509_free(x509);
+    X509_NAME_free(name);
     EVP_PKEY_CTX_free(evpctx);
     BIO_free(keybio);
     BIO_free(certbio);

